funnel/models/commentvote.py

```diff
@@ -111,7 +116,7 @@ def delete(self):
         Delete this comment.
         """
         if len(self.children) > 0:
-            self.status = COMMENTSTATUS.DELETED
+            self._state = COMMENTSTATUS.DELETED
```

Comment:
Never mutate `_state`. Add a `soft_delete` transition instead, and call it.
```python
@state.transition(None, state.DELETED)
def soft_delete(self):
    self.user = None
    self.message = ''
```

Reply:
Why are we deleting user and message? doesn't that cause loss of data in case if we want to un-delete or see history?

Reply:
Oh, never mind. It's an existing logic. Let me figure that out.